genai-perf/genai_perf/config/generate/search_parameters.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from math import log2
+from typing import Any, Dict, List, Optional
+
+from genai_perf.config.generate.search_parameter import (
+    ParameterCategory,
+    ParameterList,
+    ParameterUsage,
+    SearchParameter,
+)
+from genai_perf.config.input.config_command import ConfigCommand, Range
+from genai_perf.exceptions import GenAIPerfException
+
+
+class SearchParameters:
```

Comment:
2 classes that only differ because of pluralization feels like a easy way to confuse the two.
This may be the best name, I dont know. I do know that I will be confused by this in the future.

Reply:
Yeah, this was brought up in the MA review as well and my answer there was that no one will ever use the SearchParameter class except for SearchParameters, so there really shouldn't be any confusion in the future.

Reply:
I vaguely remember that. 
Thanks!